### ISSUE TITLE
MaterialDesignTextFieldBoxTextBox not complete to spec

### ISSUE THREAD

**Issue by bebenins** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#798]
[Spec here in short gif](https://material.io/guidelines/components/text-fields.html#text-fields-text-field-boxes)

- There should be bottom border (clipped on each side) even when it's not in focus.
- There should be a shadow indicting mouse hover over.
- There should be a ripple effect when focused.
- Hint should automatically move up when focused.

**Comment by wongjiahau** [comment]
@bebenins Point 4 is actually implemented, try the code below : 
```xml
<TextBox 
	materialDesign:HintAssist.Hint="Floating Hint"
	Style="{StaticResource MaterialDesignFloatingHintTextBox}" />
```

For point 1 to point 3, I think these are the latest specification, should be implemented soon.

**Comment by wongjiahau** [comment]
@ButchersBoy Do you plan to implement this soon? Because I thought their latest spec is more user friendly, as I recently just found out that some user are actually not familiar with the single-lined textbox, they don't feel like it's a textbox.

**Comment by Keboo** [comment]
Fixed by #840
